# Introduction

Our understanding of gene expression relies on the abstraction of a gene into discrete cis-regulatory elements (CRE), or parts, based on their most prominent function.
The promoter coordinates the initiation of transcription.
Transcription then progresses until it reaches the terminator, where cleavage and (in eukaryotes) polyadenylation of the nascent RNA transcript is coordinated.
Extending the abstraction to the mature messenger RNA transcript, the coding sequence is flanked by untranslated regions (UTRs) at its 5' and  3' ends.
The 5’UTR contributes to translation regulation [@Dai2015] and the 3’UTR to transcript stability and localisation [@Naseri2020].

These abstractions have facilitated the development of large genetic toolkits that are based on the modular assembly of interchangeable parts [@Curran2013; @Guo2015; @Lee2015moclo; @Garcia2018; @Andreou2018].
Such toolkits implicitly assume that parts characterised in one context will behave similarly in other contexts, for example that a terminator will behave similarly when paired with any promoter and coding sequence.
Massive parallel reporter libraries have been used to identify and characterise these elements, with studies screening a library of promoters with a single terminator [@Sharon2012; @Guo2015], or conversely a library of terminators with a single promoter [@Shalem2015].
However, such assumptions are challenged by large-scale measurements of combinations of CREs [@Kosuri2013; @Dhillon2020].  

Computational methods for the discovery and quantification of cis-regulatory sequence motifs also tend to rely on composability.
Prominent methods for mapping CRE sequence-function relationships include predicting functional data with short sequence features, often using linear models [@Shalgi2005; @Cheng2017;@Vijayabaskar2019]; or, directly comparing the sequences of genes with similar characteristics to determine the presence of short consensus sequence motifs [@Hogan2008; @Kretz2013] using motif discovery software [@Elemento2007, @Bailey2015].
Both of these approaches make the implicit approximation that the contribution of a motif is independent of context, so that the effect of combining motifs is composed of a linear sum (on the appropriate scale) of the individual CRE contributions. 
The assumption that motifs act independently helps to find elements that have clear contributions, and to simplify a vast search space that would be made exponentially larger by accounting for CRE interactions. 
However, framing the search for CREs around independent contributions from short motifs overlooks multi-part motifs, interactions between motifs, and that motifs may be active only in specific contexts.
\enlargethispage{-65.1pt}

Here, we quantify in detail the interdependencies between CREs, focusing on terminators in budding yeast, *Saccharomyces cerevisiae*.
We construct 120 combinations of promoter (including 5’UTR), coding sequence, and terminator (including 3’UTR) modules to quantify their effects on RNA abundance and protein production.
We further quantify the effects of 3'UTR motifs by inserting or removing combinations of motifs into diverse terminators. 
In order to select suitable 3’UTR motifs, we adapt a method to quantify mRNA decay as a log-scale linear combination of codon usage and motif counts [@Cheng2017].
We explore the effects of 3’UTR sequence on transcript termination by measuring polyadenylation sites using poly(A)-anchored RNA-seq.
Our results highlight limits to the composability of cis-regulatory elements, that are important to consider when designing synthetic genes and reporter assays.
