---
title: "Poly(A) Site Usage Plots"
output: 
  html_document:
    toc: true
---

# Summary

Plot poly(A)-site usage of selected genomic and plasmid mRNAs, from 2 poly(A)-anchored sequencing assays.

- QuantSeq batch 1 (modified terminators)
- QuantSeq batch 2 (additional native terminators)
- 5PSeq (a few modified terminators)

```{r setup, include=FALSE}
library(here)
library(readr)
library(tidyr)
library(dplyr)
library(latex2exp)
library(extrafont)
library(purrr)
library(cowplot)
library(stringr)
library(ggtext)
library(glue)
library(broom)

knitr::opts_chunk$set(echo = FALSE, message = FALSE, warning = FALSE)
source(here("raw_data_analysis/code/shared_figure_formatting.R"))
```

# Setup

## Define genomic gene dictionary

```{r definitions}
genomic_gene_dictionary <- tibble(construct_name = c("chrXIII", "chrIII", 
                                                     "chrXIV", "chrII", "chrVII"),
                                  promoter = c("pTSA1", "pPGK1", 
                                               "pRPS3", "pHSP26", "pPMA1"),
                                  terminator = c("tTSA1", "tPGK1",
                                                 "tRPS3", "tHSP26", "tPMA1"),
                                  mod = "genomic")

construct_stop_codon_dictionary <- read_csv(here("raw_data_analysis/data/rna_seq/gene_of_interest_stop_codons.csv"))
```

## Functions to read/load read data

```{r functions}
# reads in the polyA reads bed file output by the
# find_polyA_site.sh bash script in the nextflow_paired_reads_pipeline
# github repo. 
# For each sample file the function expects reads from three genomic genes:
# PGK1, RPS3 and TSA1 (used for normalisation), and two plasmid genes:
# URA3 and the construct of interest.
# The three genomic genes are on different chromosomes so mapped reads 
# are easily differentiated using the first column of the bed file
# The two plasmid genes are mapped to the same construct and are differentiated
# by the read_end position: read_end > 4000 -> URA3 , read_end < 4000 ->
# construct
load_polyA_reads <- function(filename) {
  full_bed_tibble<- read_tsv(filename,
                        col_names = c("construct_name", "chromStart", "read_end", 
                                      "name", "score", "strand", "thickStart",
                                      "thickEnd", "itemRgb", "blockCount",
                                      "blockSizes", "blockStarts")) 
  if(nrow(full_bed_tibble) > 0){
    # label reads mapped to construct of interest according to plasmid name (separating URA3 genes with by read end position)
    plasmid_construct_reads <- full_bed_tibble %>%
      filter(str_detect(construct_name, "mCherry")) %>%
      select(construct_name,read_end) %>%
      mutate(terminator = str_extract(construct_name, "t[A-Z0-9]+")) %>%
      mutate(promoter = str_extract(construct_name, "p[A-Z0-9]+")) %>%
      mutate(mod = str_extract(construct_name, "[a-zA-Z0-9]+$")) %>%
      mutate(sample = str_extract(filename, "(?<=/)(WD_)?[ABCEdt0-9]+(?=_)")) %>%
      mutate(URA3 = (read_end > 4000))
    
    # label reads mapped to genomic genes according to chromosome
    genomic_reads <- full_bed_tibble %>%
      filter(!str_detect(construct_name, "mCherry")) %>%
      select(construct_name,read_end) %>%
      inner_join(genomic_gene_dictionary) %>%
      mutate(sample = str_extract(filename, "(?<=/)(WD_)?[ABCEdt0-9]+(?=_)")) %>%
      mutate(URA3 = FALSE)
    
    bind_rows(plasmid_construct_reads, genomic_reads) 
  }
  else{NULL}
}

# for nucleotides without count data fill with 0 or value of last nucleotide with count data

fill_missing_nucleotide_data <- function(nucleotide_with_known_counts, empty_full_plasmid){
  # if first nucleotide has no entry, make a cumulative count 0 entry
  first_entry <- nucleotide_with_known_counts %>%
    filter(read_end == min(empty_full_plasmid$read_end))
  
  if(nrow(first_entry) == 0){
    nucleotide_with_known_counts <- nucleotide_with_known_counts[1,] %>%
      mutate(read_end = min(empty_full_plasmid$read_end),
             cumulative_counts = 0,
             rel_cumulative_counts = 0) %>%
      bind_rows(nucleotide_with_known_counts)
  }
  
  
  empty_full_plasmid %>%
    filter(construct_name == nucleotide_with_known_counts %>% pull(construct_name) %>% unique) %>%
  left_join(nucleotide_with_known_counts) %>%
  fill(read_end, terminator, promoter, mod, sample, URA3, cumulative_counts, rel_cumulative_counts, .direction = "down")
}

calc_polyA_site_usage <- function(polyA_count_data, polyA_site_position){
  total_counts_pre_polyA <- polyA_count_data %>%
    filter(read_end == polyA_site_position) %>%
    pull(cumulative_counts)
    
  
  total_counts_post_polyA <- polyA_count_data %>%
    filter(read_end == max(read_end)) %>%
    pull(cumulative_counts)
  
  tibble(sample = unique(polyA_count_data$sample), mod = unique(polyA_count_data$mod), pre_polyA_site_count_ratio = total_counts_pre_polyA / total_counts_post_polyA)
}
```

## Functions to help with making plots

```{r plot-functions}
cumulative_reads_plot_theme <- 
  list(labs(x = "Position relative to stop codon",
            y="Cumulative fraction of poly(A) reads"),
       guides(linetype = "none",
              shape = "none",
              colour = "none"),
       scale_y_continuous(expand = expansion(add = c(0.001,0.05)),
                          labels = c("0", "", "0.5", "", "1")),
       theme(legend.position = "bottom",
             panel.border = element_blank()))

cumulative_reads_plot_RPS3_TSA1  <- 
  list(geom_step(aes(x = read_end_gap,
                  y = rel_cumulative_counts,
                  linetype = promoter,
                  colour = label,
                  group = sample)),
    scale_colour_manual("Construct", values = RPS3_TSA1_colour_scheme),
    scale_linetype_manual("Promoter", values = c("pRPS3" = "solid", 
                                                 "pPGK1" = "dashed", 
                                                 "pTSA1" = "dotdash")),
    scale_shape_discrete("Promoter", drop = FALSE))

cumulative_reads_plot_genomic_vs_plasmid  <- 
  list(geom_step(aes(x = read_end,
                     y = rel_cumulative_counts,
                     colour = promoter,
                     group = interaction(sample, mod),
                     linetype = mod)),
       scale_y_continuous(name = "Cumulative fraction of poly(A) reads",
                          expand = expansion(add = c(0.001,0.05)),
                          labels = c("0", "", "0.5", "", "1")),
       scale_colour_manual("Promoter", 
                        breaks = c("pPGK1", "pTSA1", "pRPS3", "pRPS13", "pHSP26", "pTOS6", "pPMA1"),
                        values = brewer_pal(type = "qual", palette = "Set2")(7)),
       scale_linetype_manual("Location", 
                             values = c("genomic" = "solid", 
                                        "WT" = "dashed"),
                              labels = c("genomic" = "Genome", 
                                        "WT" = "Plasmid" )),
       guides(colour = "none", linetype = "none"),
       labs(x = "Position relative to stop codon"),
       theme(#plot.title = element_text(hjust = 0.01, vjust = -4),
             plot.title.position = "panel",
             plot.margin = margin(0.1, 0.1, 0.1, 0.1, "cm")))


make_scale_x_terminator <- function(tstart = 0, tstop = 200, ...) {
  # convenience function for setting x axis scale for cumulative reads
  scale_x_continuous(limits = c(tstart, tstop),
                     expand = expansion(add = c(1,1)),
                     ...) 
}

# set scale for each terminator so only define once
scale_x_tRPS3  <- make_scale_x_terminator(30, 80)
scale_x_tPGK1  <- make_scale_x_terminator(70, 170)
scale_x_tTSA1  <- make_scale_x_terminator(80, 120)
scale_x_tHSP26 <- make_scale_x_terminator(125, 250)
scale_x_tPMA1  <- make_scale_x_terminator(0, 500)
scale_x_tTOS6  <- make_scale_x_terminator(0, 300)
scale_x_tRPS3_mod  <- make_scale_x_terminator(47, 91)
scale_x_tTSA1_mod  <- make_scale_x_terminator(110, 150)


construct_sequence_plots <- list(theme(axis.ticks = element_blank(),
        axis.title.y = element_blank(),
        panel.grid.major.y = element_blank(),
        panel.border = element_blank(),
        axis.line.y = element_blank()),
  labs(x="Position relative to stop codon"))

```

## Load poly(A) site reads

```{r load_polyA_reads, warnings = FALSE, message = FALSE, include=FALSE}
all_QuantSeq_PolyA_read_files <- list.files(path = here("raw_data_analysis/data/rna_seq/construct_polyA_bam/QuantSeq/"), pattern = ".bed$", recursive = TRUE, full.names = TRUE)

all_QuantSeq_batch2_PolyA_read_files <- list.files(path = here("raw_data_analysis/data/rna_seq/construct_polyA_bam/QuantSeq_batch2/"), pattern = ".bed$", recursive = TRUE, full.names = TRUE)

all_5PSeq_PolyA_read_files <- list.files(path = here("raw_data_analysis/data/rna_seq/construct_polyA_bam/5PSeq/"), pattern = ".bed$", recursive = TRUE, full.names = TRUE)

full_5PSeq_PolyA_reads <- all_5PSeq_PolyA_read_files %>%
  lapply(load_polyA_reads) %>%
  purrr::reduce(bind_rows)

full_QuantSeq_PolyA_reads <- all_QuantSeq_PolyA_read_files %>%
  lapply(load_polyA_reads) %>%
  purrr::reduce(bind_rows)

full_QuantSeq_batch2_PolyA_reads <- all_QuantSeq_batch2_PolyA_read_files %>%
  lapply(load_polyA_reads) %>%
  purrr::reduce(bind_rows) %>%
  mutate(mod = ifelse(str_detect(construct_name,"mCherry"),"WT","genomic"),
         promoter = ifelse((construct_name == "chrXIV" & read_end < 100000), 
                           "pTOS6", ifelse((construct_name == "chrXIV" & read_end > 100000),
                                           "pRPS3", promoter)),
         terminator = ifelse((construct_name == "chrXIV" & read_end < 100000), 
                           "tTOS6", ifelse((construct_name == "chrXIV" & read_end > 100000),
                                           "tRPS3", terminator)))

```


## Calculate cumulative read distributions on select genes


```{r calculate-cumulative-distibution, include=FALSE}

# Filter read_end < 300 because genes of interest have shorter 3'UTRs

full_QuantSeq_polyA_reads_from_stop_codon <- 
  full_QuantSeq_PolyA_reads %>%
  inner_join(construct_stop_codon_dictionary) %>%
  mutate(read_end = read_end - stop_codon_pos) %>%
  filter(read_end > 0, read_end < 300)

full_5PSeq_polyA_reads_from_stop_codon <- full_5PSeq_PolyA_reads %>%
  filter(!sample %in% c("12dt", "14dt", "16dt")) %>%
  inner_join(construct_stop_codon_dictionary) %>%
  mutate(read_end = read_end - stop_codon_pos) %>%
  filter(read_end > 0, read_end < 300)

full_QuantSeq_batch2_polyA_reads_from_stop_codon <- full_QuantSeq_batch2_PolyA_reads  %>%
  inner_join(construct_stop_codon_dictionary) %>%
  mutate(read_end = read_end - stop_codon_pos) %>%
  mutate(read_end = ifelse(promoter == "pPMA1", - read_end, read_end)) %>%
  filter(read_end > 0)




full_QuantSeq_cumulative_polyA_reads <- full_QuantSeq_polyA_reads_from_stop_codon %>%
  group_by(construct_name,read_end, terminator, promoter, mod,sample, URA3) %>%
  summarise(read_count = n()) %>%
  arrange(construct_name, read_end) %>%
  group_by(construct_name, URA3, sample) %>%
  mutate(cumulative_counts = cumsum(read_count), 
         total_counts = sum(read_count),
         rel_counts = read_count / total_counts,
         rel_cumulative_counts = cumulative_counts / total_counts)

full_QuantSeq_batch2_cumulative_polyA_reads <- full_QuantSeq_batch2_polyA_reads_from_stop_codon %>%
  group_by(construct_name,read_end, terminator, promoter, mod,sample, URA3) %>%
  summarise(read_count = n()) %>%
  arrange(construct_name, read_end) %>%
  group_by(construct_name, terminator, promoter, mod,sample, URA3) %>%
  mutate(cumulative_counts = cumsum(read_count), 
         total_counts = sum(read_count),
         rel_counts = read_count / total_counts,
         rel_cumulative_counts = cumulative_counts / total_counts)

full_QuantSeq_batch2_cumulative_polyA_reads_RPS3 <- full_QuantSeq_batch2_polyA_reads_from_stop_codon %>%
  filter(read_end < 300) |>
  group_by(construct_name,read_end, terminator, promoter, mod,sample, URA3) %>%
  summarise(read_count = n()) %>%
  arrange(construct_name, read_end) %>%
  group_by(construct_name, terminator, promoter, mod,sample, URA3) %>%
  mutate(cumulative_counts = cumsum(read_count), 
         total_counts = sum(read_count),
         rel_counts = read_count / total_counts,
         rel_cumulative_counts = cumulative_counts / total_counts)

full_5PSeq_cumulative_polyA_reads <- full_5PSeq_polyA_reads_from_stop_codon %>%
  group_by(construct_name,read_end, terminator, promoter, mod,sample, URA3) %>%
  summarise(read_count = n()) %>%
  arrange(construct_name, read_end) %>%
  group_by(construct_name, URA3, sample) %>%
  mutate(cumulative_counts = cumsum(read_count), 
         total_counts = sum(read_count),
         rel_counts = read_count / total_counts,
         rel_cumulative_counts = cumulative_counts / total_counts)

# Create data frames to pad out counts for every read position
empty_full_length_QuantSeq_plasmids <- tibble(construct_name = rep(full_QuantSeq_cumulative_polyA_reads %>% pull(construct_name) %>% unique(), each = 301), 
                                              read_end = rep(0:300,18))

empty_full_length_QuantSeq_batch2_plasmids <- tibble(construct_name = rep(full_QuantSeq_batch2_cumulative_polyA_reads %>% pull(construct_name) %>% unique(), each = 501), 
                                              read_end = rep(0:500,10))

empty_full_length_QuantSeq_batch2_RPS3 <- tibble(construct_name = rep(full_QuantSeq_batch2_cumulative_polyA_reads_RPS3 %>% pull(construct_name) %>% unique(), each = 301), 
                                              read_end = rep(0:300,10))

empty_full_length_5PSeq_plasmids <- tibble(construct_name = rep(full_5PSeq_cumulative_polyA_reads %>% pull(construct_name) %>% unique(), each = 301), 
                                              read_end = rep(0:300,10))


full_length_QuantSeq_plasmids_cumulative_polyA_reads <- full_QuantSeq_cumulative_polyA_reads %>%
  select(-read_count, - total_counts) %>%
  group_by(construct_name, sample, URA3) %>%
  do(fill_missing_nucleotide_data(., empty_full_length_QuantSeq_plasmids)) %>%
  inner_join(construct_to_label_dictionary_TSA1_RPS3, by = c("mod" = "construct")) %>%
  mutate(label = factor(label, levels = construct_to_label_dictionary_TSA1_RPS3$label),
         promoter = factor(promoter, levels = c("pPGK1", "pRPS3", "pTSA1")))

full_length_QuantSeq_batch2_plasmids_cumulative_polyA_reads <- full_QuantSeq_batch2_cumulative_polyA_reads %>%
  select(-read_count, - total_counts) %>%
  group_by(construct_name, sample, URA3, terminator, promoter, mod) %>%
  do(fill_missing_nucleotide_data(., empty_full_length_QuantSeq_batch2_plasmids)) %>%
  mutate(label = factor(mod),
         promoter = factor(promoter, levels = c("pHSP26","pRPS13","pPGK1", "pRPS3", "pTSA1","pTOS6","pPMA1")))

full_length_QuantSeq_batch2_plasmids_cumulative_polyA_reads_RPS3 <- full_QuantSeq_batch2_cumulative_polyA_reads_RPS3 %>%
  select(-read_count, - total_counts) %>%
  group_by(construct_name, sample, URA3, terminator, promoter, mod) %>%
  do(fill_missing_nucleotide_data(., empty_full_length_QuantSeq_batch2_RPS3)) %>%
  mutate(label = factor(mod),
         promoter = factor(promoter, levels = c("pHSP26","pRPS13","pPGK1", "pRPS3", "pTSA1","pTOS6","pPMA1")))



full_length_5PSeq_plasmids_cumulative_polyA_reads <- full_5PSeq_cumulative_polyA_reads %>%
  select(-read_count, - total_counts) %>%
  group_by(construct_name, sample, URA3) %>%
  do(fill_missing_nucleotide_data(., empty_full_length_5PSeq_plasmids)) %>%
  inner_join(construct_to_label_dictionary_TSA1_RPS3, by = c("mod" = "construct")) %>%
  mutate(label = factor(label, levels = construct_to_label_dictionary_TSA1_RPS3$label))

full_length_QuantSeq_genomic_cumulative_polyA_reads <- full_QuantSeq_cumulative_polyA_reads %>%
  select(-read_count, - total_counts) %>%
  group_by(construct_name, sample) %>%
  do(fill_missing_nucleotide_data(., empty_full_length_QuantSeq_plasmids)) %>%
  filter(mod == "genomic")

full_length_5PSeq_genomic_cumulative_polyA_reads <- full_5PSeq_cumulative_polyA_reads %>%
  select(- read_count, - total_counts) %>%
  group_by(construct_name, sample) %>%
  do(fill_missing_nucleotide_data(., empty_full_length_5PSeq_plasmids)) %>%
  filter(mod == "genomic")

```

# Plot cumulative read fractions for native genes, as quality control

```{r plot_QC_cumulative_graph}
## Create 5PSeq QC cumulative plots by comparing genomic RPS3, PGK1 and TSA1 terminators across samples

full_length_QuantSeq_batch2_plasmids_cumulative_QC <- full_length_QuantSeq_batch2_plasmids_cumulative_polyA_reads %>%
  filter(mod == "genomic")

full_length_QuantSeq_plasmids_cumulative_QC <- full_QuantSeq_cumulative_polyA_reads %>%
  filter(mod == "genomic") %>%
  select(-read_count, - total_counts) %>%
  group_by(construct_name, sample, URA3) %>%
  do(fill_missing_nucleotide_data(., empty_full_length_QuantSeq_plasmids))

genomic_QuantSeq_cumulative_QC <- 
  ggplot(full_QuantSeq_cumulative_polyA_reads %>%
           filter(mod == "genomic", read_end <= 200)) +
  geom_step(aes(x = read_end, y = rel_cumulative_counts, colour = sample)) +
  labs(x="Position relative to stop codon", y="Cumulative fraction of reads", title = "Genomic, QuantSeq") +
  cumulative_reads_plot_theme + 
  facet_wrap(~terminator, ncol = 1, scales = "free_x") +
  expand_limits(x = 0) + 
  theme(legend.position = "bottom")

## Create 5PSeq QC cumulative plots

genomic_5PSeq_cumulative_QC <- 
  ggplot(full_5PSeq_cumulative_polyA_reads %>%
           filter(mod == "genomic", read_end <= 200)) +
  geom_step(aes(x = read_end, y = rel_cumulative_counts, colour = sample)) +
  labs(x="Position relative to stop codon", y="Cumulative fraction of reads", title = "Genomic, 5PSeq") +
  # make_scale_x_terminator(tstop = 200) +
  cumulative_reads_plot_theme + 
  facet_wrap(~terminator, ncol = 1, scales = "free_x") +
  expand_limits(x = 0) +
  theme(legend.position = "bottom")

genomic_QuantSeq_combined_cumulative_QC <- 
  plot_grid(genomic_QuantSeq_cumulative_QC,
            genomic_5PSeq_cumulative_QC, 
            ncol = 2)

genomic_QuantSeq_combined_cumulative_QC

# Repeat for QuantSeq batch2
genomic_QuantSeq_batch2_cumulative_QC <- 
ggplot(full_length_QuantSeq_batch2_plasmids_cumulative_polyA_reads %>%
         filter(mod == "genomic")) +
  geom_step(aes(x = read_end, y = rel_cumulative_counts, colour = sample)) +
  labs(x="Position relative to stop codon", y="Cumulative fraction of reads", title = "Genomic, QuantSeq batch 2") +
  make_scale_x_terminator(tstop = 500) +
  cumulative_reads_plot_theme + 
  facet_wrap(~terminator, ncol = 1) +
  theme(legend.position = "bottom")
genomic_QuantSeq_batch2_cumulative_QC
```

```{r output-polyA-genomic-QC-plot}

ggsave2(here("raw_data_analysis/figures/polya_genomic_QC_plot_QuantSeq_and_5PSeq.png"), 
        genomic_QuantSeq_combined_cumulative_QC, 
        width = fig_width_2column,
        height = 150,
        units = "mm",
        dpi = fig_dpi,
        bg = "white")

ggsave2(here("raw_data_analysis/data/figures/polya_QC_plot_QuantSeq_batch2.png"), 
        genomic_QuantSeq_batch2_cumulative_QC, 
        width = fig_width_2column,
        height = 150,
        units = "mm",
        dpi = fig_dpi,
        bg = "white")


```

# Plot plasmid vs genome poly(A) sites for QuantSeq data

```{r plot-QC-cumulative-graph-plasmid-WT-vs-genome}
# Collect the data
plasmid_vs_genome_polyA_QuantSeq <- 
  full_length_QuantSeq_plasmids_cumulative_polyA_reads %>%
  filter(mod == "WT", !URA3) %>%
  bind_rows(full_length_QuantSeq_plasmids_cumulative_QC %>%
              filter(terminator %in% c("tRPS3", "tTSA1"),
                     sample %in% c("A12", "B12", "A1", "B1", "A5", "B5"))) %>%
  bind_rows(full_length_QuantSeq_batch2_plasmids_cumulative_polyA_reads %>%
              filter(!URA3) ) %>%
  mutate(promoter = factor(promoter, levels = c("pPGK1", "pTSA1", "pRPS3", "pRPS13", "pHSP26", "pTOS6", "pPMA1")),
         terminator = factor(terminator, levels = c("tPGK1", "tTSA1", "tRPS3", "tRPS13", "tHSP26", "tTOS6", "tPMA1")))

# make a faceted plot with all the data on same x-axis,
# except exclude tPMA1 because beyond the limit of confident detection
plasmid_vs_genome_polyA_facet_plot <- 
  ggplot(plasmid_vs_genome_polyA_QuantSeq %>% 
           filter(terminator != "tPMA1")) +
  cumulative_reads_plot_genomic_vs_plasmid +
  make_scale_x_terminator(tstop = 300) + 
  facet_wrap(~terminator, ncol = 1) +
  guides(linetype = guide_legend(ncol = 1),
         colour   = guide_legend(ncol = 1))

plasmid_vs_genome_polyA_facet_plot
```

```{r output-cumulative-graph-plasmid-WT-vs-genome}
ggsave2(here("results_chapter/figures/plasmid_vs_genome_polyA_combined_plot.png"), 
        plasmid_vs_genome_polyA_facet_plot,
        width = fig_width_1column,
        height = 125,
        units = "mm",
        dpi = fig_dpi,
        bg = "white")
```

# Plot plasmid vs genome poly(A) sites in more detail

Detail includes, only in matched samples.

```{r plot-QC-cumulative-graphs-plasmid-WT-vs-genome-zoomed}
plasmid_vs_genome_polyA_QuantSeq_RPS3_plot <- 
  ggplot(plasmid_vs_genome_polyA_QuantSeq %>%
         filter(terminator == "tRPS3",
                sample %in% c("A12", "B12", "A1", "B1"))) +
  cumulative_reads_plot_genomic_vs_plasmid + 
  labs(title = "tRPS3") +
  scale_x_tRPS3


plasmid_vs_genome_polyA_QuantSeq_TSA1_plot <- 
  ggplot(plasmid_vs_genome_polyA_QuantSeq %>%
         filter(terminator == "tTSA1",
                sample %in% c("A5", "B5"))) +
  cumulative_reads_plot_genomic_vs_plasmid + 
  labs(title = "tTSA1") +
  scale_x_tTSA1

plasmid_vs_genome_polyA_QuantSeq_HSP26_plot <- 
  ggplot(plasmid_vs_genome_polyA_QuantSeq %>%
         filter(terminator == "tHSP26",
                sample %in% c("WD_C9", "WD_C10", "WD_C11", 
                              "WD_C12", "WD_C13", "WD_C14", 
                              "WD_C15", "WD_C16"))) +
  cumulative_reads_plot_genomic_vs_plasmid + 
  labs(title = "tHSP26") +
  scale_x_tHSP26

plasmid_vs_genome_polyA_QuantSeq_TOS6_plot <- 
  ggplot(plasmid_vs_genome_polyA_QuantSeq %>%
         filter(terminator == "tTOS6",
                sample %in% c("WD_C7", "WD_C8"))) +
  cumulative_reads_plot_genomic_vs_plasmid +
  labs(title = "tTOS6") +
  scale_x_tTOS6

plasmid_vs_genome_polyA_QuantSeq_PMA1_plot <- 
  ggplot(plasmid_vs_genome_polyA_QuantSeq %>%
         filter(terminator == "tPMA1",
                sample %in% c("WD_E2", "WD_E3"))) +
  cumulative_reads_plot_genomic_vs_plasmid +
  labs(title = "tPMA1") +
  scale_x_tPMA1

# make a combined plot here, with no plot legend.

plot_grid(plasmid_vs_genome_polyA_QuantSeq_TSA1_plot +
            theme(axis.title.x = element_blank(), axis.title.y = element_blank()),
          plasmid_vs_genome_polyA_QuantSeq_RPS3_plot +
            theme(axis.title.x = element_blank(), axis.title.y = element_blank()),
          plasmid_vs_genome_polyA_QuantSeq_HSP26_plot +
            theme(axis.title.x = element_blank()),
          plasmid_vs_genome_polyA_QuantSeq_TOS6_plot +
            theme(axis.title.x = element_blank(), axis.title.y = element_blank()),
          plasmid_vs_genome_polyA_QuantSeq_PMA1_plot +
            theme(axis.title.x = element_blank(), axis.title.y = element_blank()),
          ggdraw() + 
            draw_label("Position relative to stop codon", size = 12),
          ncol = 1,
          align = "v",
          rel_heights = c(1,1,1,1,1,0.3)
)
```


# Plot plasmid vs genome poly(A) sites from 5PSeq data

```{r plasmid_vs_genome_polyA_5PSeq}

plasmid_vs_genome_polyA_5PSeq <- 
  full_length_5PSeq_plasmids_cumulative_polyA_reads %>%
  filter(mod == "WT", !URA3) %>%
  bind_rows(full_length_5PSeq_genomic_cumulative_polyA_reads %>%
              filter(terminator %in% c("tRPS3", "tTSA1"),
                     sample %in% c("3dt", "4dt")))

plasmid_vs_genome_polyA_5PSeq_RPS3_plot <- 
  ggplot(plasmid_vs_genome_polyA_5PSeq %>%
           filter(terminator == "tRPS3", !URA3)) +
  geom_step(aes(x = read_end,
                y = rel_cumulative_counts,
                colour = sample,
                linetype = mod)) +
  labs(x="Position relative to stop codon", y="Cumulative\nfrac. of reads", 
       title = "5PSeq, tRPS3") +
  make_scale_x_terminator(tstop = 200) +
  guides(colour = "none") +
  scale_linetype_manual("Location", 
                        values = c("genomic" = "solid", 
                                   "WT" = "dashed"),
                        labels = c("genomic" = "Genome", 
                                   "WT" = "Plasmid" )) +
  scale_y_continuous(expand = expansion(add = c(0.001,0.05)),
                     labels = c("0", "", "0.5", "", "1")) +
  theme(panel.border = element_blank())

plasmid_vs_genome_polyA_5PSeq_RPS3_plot
```

#

```{r output-polyA-QC-plot}

ggsave2(here("supplementary_data_chapter/figures/polya_QC_plot_QuantSeq_and_5PSeq.png"), 
        plot_grid(genomic_QuantSeq_combined_cumulative_QC,
                  plot_grid(plasmid_vs_genome_polyA_5PSeq_RPS3_plot, 
                            NULL, 
                            nrow = 1, rel_widths = c(2,1)),
                  ncol = 1, rel_heights = c(12,5),
                  labels = "AUTO"),
        width = fig_width_2column,
        height = 170,
        units = "mm",
        dpi = fig_dpi,
        bg = "white")

```

# Plot cumulative reads from QuantSeq for modified terminators

```{r plot-cumulative-graph}
cumulative_graph_QuantSeq_pRPS3 <- ggplot(full_length_QuantSeq_plasmids_cumulative_polyA_reads %>%
  filter(promoter == "pRPS3", mod != "genomic", !URA3)) +
  geom_line(aes(x = read_end, y = rel_cumulative_counts, linetype = str_detect(sample, "A"), group = sample, colour = label)) +
  guides(linetype = FALSE, colour = FALSE) +
  labs(x="", y="Fraction of Reads", title = "pRPS3-tRPS3 (QuantSeq)") +
  scale_colour_manual(values = RPS3_TSA1_colour_scheme)

cumulative_graph_QuantSeq_batch2<- ggplot(full_length_QuantSeq_batch2_plasmids_cumulative_polyA_reads %>%
  filter(mod != "genomic", !URA3)) +
  geom_line(aes(x = read_end, y = rel_cumulative_counts, group = sample, colour = paste(promoter, terminator))) +
  guides(linetype = FALSE) +
  labs(x="", y="Fraction of Reads", title = "QuantSeq Heat Shock")

cumulative_graph_QuantSeq_pTSA1 <- ggplot(full_length_QuantSeq_plasmids_cumulative_polyA_reads %>%
  filter(promoter == "pTSA1", mod != "genomic", !URA3)) +
  geom_line(aes(x = read_end, y = rel_cumulative_counts, linetype = str_detect(sample, "A"), group = sample, colour = label)) +
  guides(colour = FALSE)  +
  labs(x="Position relative to stop codon", y="", title = "pTSA1-tTSA1 (QuantSeq)") +
  scale_linetype_discrete(labels = c(1,2), name = "Rep") +
  scale_colour_manual(values = RPS3_TSA1_colour_scheme)

cumulative_graph_QuantSeq_pPGK1 <- ggplot(full_length_QuantSeq_plasmids_cumulative_polyA_reads %>%
  filter(promoter == "pPGK1", mod != "genomic", !URA3)) +
  geom_line(aes(x = read_end, y = rel_cumulative_counts, linetype = str_detect(sample, "A"), group = sample, colour = label)) +
  guides(colour = FALSE, linetype = FALSE)  +
  labs(x="Position relative to stop codon", y="Fraction of Reads", title = "pPGK1-tRPS3 (QuantSeq)") +
  scale_colour_manual(values = RPS3_TSA1_colour_scheme)

cumulative_graph_QuantSeq_norm_genes <- ggplot(full_length_QuantSeq_genomic_cumulative_polyA_reads) +
  geom_line(aes(x = read_end, y = rel_cumulative_counts, linetype = sample, colour = promoter)) +
  guides(linetype = FALSE)  +
  labs(x="Position relative to stop codon", y="Fraction of Reads", title = "Normalising Genes (QuantSeq)")

cumulative_graph_5PSeq_pRPS3 <- ggplot(full_length_5PSeq_plasmids_cumulative_polyA_reads %>%
  filter(promoter == "pRPS3", mod != "genomic", !URA3)) +
  geom_line(aes(x = read_end, y = rel_cumulative_counts, linetype = str_detect(sample, "A"), group = sample, colour = label)) +
  guides(linetype = FALSE, colour = FALSE) +
  labs(x="", y="Fraction of Reads", title = "pRPS3-tRPS3 (5PSeq)") +
  scale_colour_manual(values = RPS3_TSA1_colour_scheme)

cumulative_graph_5PSeq_norm_genes <- ggplot(full_length_5PSeq_genomic_cumulative_polyA_reads) +
  geom_line(aes(x = read_end, y = rel_cumulative_counts, linetype = sample, colour = promoter)) +
  guides(linetype = FALSE)  +
  labs(x="Position relative to stop codon", y="Fraction of Reads", title = "Normalising Genes (5PSeq)")


```

# Normalise transcript counts


```{r normalise-to-median-norm-gene}
transcript_counts_QuantSeq <- full_QuantSeq_PolyA_reads %>%
  group_by(promoter, terminator, mod, sample, URA3) %>%
  summarise(count = n())

median_norm_gene_counts_QuantSeq <- transcript_counts_QuantSeq %>%
  filter(mod == "genomic" | URA3) %>%
  group_by(sample) %>%
  summarise(norm_gene_count = median(count))

norm_transcript_counts_QuantSeq <- transcript_counts_QuantSeq %>%
  inner_join(median_norm_gene_counts_QuantSeq) %>%
  mutate(norm_count = count / norm_gene_count) %>%
  filter(!URA3, mod != "genomic") %>%
  select(-URA3)

mean_norm_modNNN_counts_QuantSeq <- norm_transcript_counts_QuantSeq %>%
  filter(mod == "mod0") %>%
  group_by(promoter) %>%
  summarise(mean_modNNN = mean(norm_count))

construct_fold_change_QuantSeq_mod_NNN <- norm_transcript_counts_QuantSeq %>%
  inner_join(mean_norm_modNNN_counts_QuantSeq) %>%
  mutate(rel_abund_delta_deltacq = norm_count / mean_modNNN) %>%
  inner_join(construct_to_label_dictionary_TSA1_RPS3, by = c("mod" = "construct")) %>%
  dplyr::rename("mod_label" = "label")

ggplot(construct_fold_change_QuantSeq_mod_NNN) +
  geom_point(aes(y = mod, x = rel_abund_delta_deltacq, colour = str_detect(sample, "A"))) +
  facet_wrap(~promoter) +
  guides(colour = FALSE) 

transcript_counts_5PSeq <- full_5PSeq_PolyA_reads %>%
  group_by(promoter, terminator, mod, sample, URA3) %>%
  summarise(count = n())

median_norm_gene_counts_5PSeq <- transcript_counts_5PSeq %>%
  filter(mod == "genomic" | URA3) %>%
  group_by(sample) %>%
  summarise(norm_gene_count = median(count))

norm_transcript_counts_5PSeq <- transcript_counts_5PSeq %>%
  inner_join(median_norm_gene_counts_5PSeq) %>%
  mutate(norm_count = count / norm_gene_count) %>%
  filter(!URA3, mod != "genomic") %>%
  select(-URA3)

mean_norm_modNNN_counts_5PSeq <- norm_transcript_counts_5PSeq %>%
  filter(mod == "mod0") %>%
  group_by(promoter) %>%
  summarise(mean_modNNN = mean(norm_count))

construct_fold_change_5PSeq_mod_NNN <- norm_transcript_counts_5PSeq %>%
  inner_join(mean_norm_modNNN_counts_5PSeq) %>%
  mutate(rel_abund_delta_deltacq = norm_count / mean_modNNN) %>%
  inner_join(construct_to_label_dictionary_TSA1_RPS3, by = c("mod" = "construct")) %>%
  dplyr::rename("mod_label" = "label")

ggplot(construct_fold_change_5PSeq_mod_NNN) +
  geom_point(aes(y = mod, x = rel_abund_delta_deltacq, colour = str_detect(sample, "A"))) +
  facet_wrap(~promoter) +
  guides(colour = FALSE)

```

# Import RT-qPCR results for modified terminators


```{r import_qPCR_results}
TSA1_deltadeltacq <- read_csv(here("raw_data_analysis/data/norm_qpcr/motif_context_dependence/pTSA1_pPGK1_pSRO9_tTSA1_deltadeltacq_platesnorm_summarise.csv")) %>%
  mutate(mod_label = factor(construct,
                       levels = construct_to_label_dictionary_TSA1_RPS3$construct,
                       labels = construct_to_label_dictionary_TSA1_RPS3$label),
         promoter = factor(promoter, levels = c("pTSA1","pPGK1","pSRO9")))

RPS3_deltadeltacq <- read_csv(here("raw_data_analysis/data/norm_qpcr/motif_context_dependence/pRPS3_pPGK1_pSRO9_tRPS3_deltadeltacq_platesnorm_summarise.csv")) %>%
  mutate(mod_label = factor(construct,
                       levels = construct_to_label_dictionary_TSA1_RPS3$construct,
                       labels = construct_to_label_dictionary_TSA1_RPS3$label),
         promoter = factor(promoter, levels = c("pRPS3","pPGK1","pSRO9")))

TSA1_deltadeltacq_plot <- ggplot(data = TSA1_deltadeltacq)+
  RNA_relative_abundance_figure_options +
  scale_colour_manual(values=RPS3_TSA1_colour_scheme) +
  theme(axis.text.y=element_text(colour=RPS3_TSA1_colour_scheme, face = "bold")) +
  labs(x ="Fold change in RNA abundance relative to tTSA1 mod_NNN", y = "") +
  facet_wrap(~promoter,ncol = 3) 

RPS3_deltadeltacq_plot <- ggplot(data = RPS3_deltadeltacq) +
  RNA_relative_abundance_figure_options +
  scale_colour_manual(values=RPS3_TSA1_colour_scheme) +
  theme(axis.text.y=element_text(colour=RPS3_TSA1_colour_scheme, face = "bold")) +
  labs(x ="Fold change in RNA abundance relative to tRPS3 mod_NNN", y = "") +
  facet_wrap(~promoter,ncol = 3) 

```

# Compare RT-qPCR results with QuantSeq results

```{r compare_qPCR_and_RNAseq}
mean_qpcr_vs_QuantSeq_dataset_mod_NNN <- construct_fold_change_QuantSeq_mod_NNN %>%
              mutate(source = "RNA-Seq", type = "QuantSeq") %>%
  bind_rows(bind_rows(TSA1_deltadeltacq %>% filter(promoter == "pTSA1"),
                       RPS3_deltadeltacq%>% filter(promoter %in% c("pRPS3",
                                                                   "pPGK1"))) %>% 
              select(promoter, terminator, mod_label, rel_abund_delta_deltacq) %>%
              mutate(source = "qPCR", type = "QuantSeq") %>%
              filter(mod_label %in% c("mod_NNN","mod_NAA","WT", 
                                      "mod_HTH","mod_NTN"))) %>%
  group_by(promoter, source, mod_label, terminator, type) %>%
  summarise(mean_rel_abund = mean(rel_abund_delta_deltacq)) %>%
  ungroup() %>%
  pivot_wider(names_from = "source", values_from = "mean_rel_abund")

mean_qpcr_vs_5PSeq_dataset_mod_NNN <- construct_fold_change_5PSeq_mod_NNN %>%
              mutate(source = "RNA-Seq", type = "5PSeq") %>%
  bind_rows(bind_rows(RPS3_deltadeltacq%>% filter(promoter %in% c("pRPS3"))) %>% 
              select(promoter, terminator, mod_label, rel_abund_delta_deltacq) %>%
              mutate(source = "qPCR", type = "5PSeq") %>%
              filter(mod_label %in% c("mod_NNN","mod_NAA","WT", 
                                      "mod_HTH","mod_NTN","mod_HNH","mod_NGG"))) %>%
  group_by(promoter, source, mod_label, terminator, type) %>%
  summarise(mean_rel_abund = mean(rel_abund_delta_deltacq)) %>%
  ungroup() %>%
  pivot_wider(names_from = "source", values_from = "mean_rel_abund")

qpcr_vs_QuantSeq_dataset_R_values <-mean_qpcr_vs_QuantSeq_dataset_mod_NNN %>% 
  group_by(promoter) %>% 
  summarise(r_cor = cor(qPCR, `RNA-Seq`)) %>%
  mutate(r_cor_label = paste(TeX(paste0("R = ", signif(r_cor,digits = 3)))))

qPCR_vs_RNASeq_plot_QuantSeq <- ggplot(mean_qpcr_vs_QuantSeq_dataset_mod_NNN ) +
  geom_point(aes(x = `RNA-Seq`, y = qPCR, colour = mod_label)) +
  geom_abline(slope = 1, intercept = 0) +
  coord_equal() +
  facet_wrap(~promoter, nrow = 1) +
  scale_colour_manual(values = RPS3_TSA1_colour_scheme) +
  guides(colour = "none") + 
  geom_text(aes(label = r_cor_label), 
            data = qpcr_vs_QuantSeq_dataset_R_values, 
            x = 2.4, y = 0.3, 
            size = text_cor_size, hjust = 1, vjust = 0)

qpcr_vs_5PSeq_dataset_R_values <- mean_qpcr_vs_5PSeq_dataset_mod_NNN %>% 
  group_by(promoter) %>% 
  summarise(r_cor = cor(qPCR, `RNA-Seq`)) %>%
  mutate(r_cor_label = paste(TeX(paste0("R = ", signif(r_cor,digits = 3)))))

qPCR_vs_RNASeq_plot_5PSeq <- ggplot(mean_qpcr_vs_5PSeq_dataset_mod_NNN ) +
  geom_point(aes(x = `RNA-Seq`, y = qPCR, colour = mod_label, shape = terminator)) +
  geom_abline(slope = 1, intercept = 0) +
  guides(colour = "none", shape = "none") +
  scale_colour_manual(values = RPS3_TSA1_colour_scheme) +
  theme(legend.box = "horizontal") +
  coord_equal() + 
  geom_text(aes(label = r_cor_label), 
            data = qpcr_vs_5PSeq_dataset_R_values, 
            x = 1.4, y = 0.3, 
            size = text_cor_size, hjust = 1, vjust = 0)

mean_qpcr_vs_QuantSeq_dataset_mod_NNN_R_squared <- mean_qpcr_vs_QuantSeq_dataset_mod_NNN %>%
  group_by(promoter) %>%
  summarise(R_squared = cor(qPCR, `RNA-Seq`)^2)

mean_qpcr_vs_5PSeq_dataset_mod_NNN_R_squared <- mean_qpcr_vs_5PSeq_dataset_mod_NNN %>%
  summarise(R_squared = cor(qPCR, `RNA-Seq`)^2)

```


# Create virtual gapped WT terminators to compare with modified insertion contructs

```{r create-motifs-gaps-in-wt}
tRPS3_motif_insertion_sites <- tibble(site = c(1,2,3), start = c(24, 48, 61), end = c(32, 56, 69))

tTSA1_motif_insertion_sites <- tibble(site = c(1,2,3), start = c(21, 50, 83), end = c(29, 58, 91))


tRPS3_WT_insertion_alpha <- full_length_QuantSeq_plasmids_cumulative_polyA_reads %>%
  mutate(read_end_gap = ifelse(mod != "WT" | terminator != "tRPS3", read_end,
                         ifelse(read_end >= tRPS3_motif_insertion_sites$start[3] - 18, read_end + 27,
                         ifelse(read_end >= tRPS3_motif_insertion_sites$start[2] - 9, read_end + 18,
                         ifelse(read_end >= tRPS3_motif_insertion_sites$start[1], read_end + 9, read_end)))))

tRPS3_WT_insertion_alpha_5PSeq <- full_length_5PSeq_plasmids_cumulative_polyA_reads %>%
  mutate(read_end_gap = ifelse(mod != "WT" | terminator != "tRPS3", read_end,
                         ifelse(read_end >= tRPS3_motif_insertion_sites$start[3] - 18, read_end + 27,
                         ifelse(read_end >= tRPS3_motif_insertion_sites$start[2] - 9, read_end + 18,
                         ifelse(read_end >= tRPS3_motif_insertion_sites$start[1], read_end + 9, read_end)))))

tRPS3_missing_values <- tRPS3_WT_insertion_alpha %>%
  filter(mod == "WT",
         terminator == "tRPS3", 
         !URA3,
         (read_end <= tRPS3_motif_insertion_sites$end[1] & 
           read_end >= tRPS3_motif_insertion_sites$start[1]) |
         (read_end <= tRPS3_motif_insertion_sites$end[2] &
           read_end >= tRPS3_motif_insertion_sites$start[2]) |
         (read_end <= tRPS3_motif_insertion_sites$end[3] &
           read_end >= tRPS3_motif_insertion_sites$start[3])) %>%
  mutate(rel_cumulative_counts = NA,
         rel_counts = NA,
         read_end_gap = read_end)

tRPS3_missing_values_5PSeq <- tRPS3_WT_insertion_alpha_5PSeq %>%
  filter(mod == "WT",
         terminator == "tRPS3", 
         !URA3,
         (read_end <= tRPS3_motif_insertion_sites$end[1] & 
           read_end >= tRPS3_motif_insertion_sites$start[1]) |
         (read_end <= tRPS3_motif_insertion_sites$end[2] &
           read_end >= tRPS3_motif_insertion_sites$start[2]) |
         (read_end <= tRPS3_motif_insertion_sites$end[3] &
           read_end >= tRPS3_motif_insertion_sites$start[3])) %>%
  mutate(rel_cumulative_counts = NA,
         rel_counts = NA,
         read_end_gap = read_end)



tRPS3_WT_insertion_gaps <- full_length_QuantSeq_plasmids_cumulative_polyA_reads %>%
  filter(mod == "WT",
         terminator == "tRPS3",
         !URA3,
         read_end >= tRPS3_motif_insertion_sites$start[1] - 1,
         read_end <= tRPS3_motif_insertion_sites$end[3]) %>%
  mutate(rel_cumulative_counts_gap = ifelse(read_end <= tRPS3_motif_insertion_sites$end[3],
                rel_cumulative_counts[read_end == tRPS3_motif_insertion_sites$start[3] - 19],
                rel_cumulative_counts)) %>%
  mutate(rel_cumulative_counts_gap = ifelse(read_end <= tRPS3_motif_insertion_sites$start[3] - 1,
                NA,
                rel_cumulative_counts_gap)) %>%
  mutate(rel_cumulative_counts_gap = ifelse(read_end <= tRPS3_motif_insertion_sites$end[2],
                rel_cumulative_counts[ read_end == tRPS3_motif_insertion_sites$start[2] - 10],
                rel_cumulative_counts_gap)) %>%
  mutate(rel_cumulative_counts_gap = ifelse(read_end <= tRPS3_motif_insertion_sites$start[2] - 1,
                NA,
                rel_cumulative_counts_gap)) %>%
  mutate(rel_cumulative_counts_gap = ifelse(read_end <= tRPS3_motif_insertion_sites$end[1],
                rel_cumulative_counts[read_end == tRPS3_motif_insertion_sites$start[1] - 1],
                rel_cumulative_counts_gap))

tRPS3_WT_insertion_gaps_5PSeq <- full_length_5PSeq_plasmids_cumulative_polyA_reads %>%
  filter(mod == "WT",
         terminator == "tRPS3",
         !URA3,
         read_end >= tRPS3_motif_insertion_sites$start[1] - 1,
         read_end <= tRPS3_motif_insertion_sites$end[3]) %>%
  mutate(rel_cumulative_counts_gap = ifelse(read_end <= tRPS3_motif_insertion_sites$end[3],
                rel_cumulative_counts[read_end == tRPS3_motif_insertion_sites$start[3] - 19],
                rel_cumulative_counts)) %>%
  mutate(rel_cumulative_counts_gap = ifelse(read_end <= tRPS3_motif_insertion_sites$start[3] - 1,
                NA,
                rel_cumulative_counts_gap)) %>%
  mutate(rel_cumulative_counts_gap = ifelse(read_end <= tRPS3_motif_insertion_sites$end[2],
                rel_cumulative_counts[ read_end == tRPS3_motif_insertion_sites$start[2] - 10],
                rel_cumulative_counts_gap)) %>%
  mutate(rel_cumulative_counts_gap = ifelse(read_end <= tRPS3_motif_insertion_sites$start[2] - 1,
                NA,
                rel_cumulative_counts_gap)) %>%
  mutate(rel_cumulative_counts_gap = ifelse(read_end <= tRPS3_motif_insertion_sites$end[1],
                rel_cumulative_counts[read_end == tRPS3_motif_insertion_sites$start[1] - 1],
                rel_cumulative_counts_gap))

tTSA1_WT_insertion_alpha <- full_length_QuantSeq_plasmids_cumulative_polyA_reads %>%
  mutate(read_end_gap = ifelse(mod != "WT" | terminator != "tTSA1", read_end,
                         ifelse(read_end >= tTSA1_motif_insertion_sites$start[3] - 18, read_end + 27,
                         ifelse(read_end >= tTSA1_motif_insertion_sites$start[2] - 9, read_end + 18,
                         ifelse(read_end >= tTSA1_motif_insertion_sites$start[1], read_end + 9, read_end)))))

tTSA1_missing_values <- tTSA1_WT_insertion_alpha %>%
  filter(mod == "WT",
         terminator == "tTSA1", 
         !URA3,
         (read_end <= tTSA1_motif_insertion_sites$end[1] & 
           read_end >= tTSA1_motif_insertion_sites$start[1]) |
         (read_end <= tTSA1_motif_insertion_sites$end[2] &
           read_end >= tTSA1_motif_insertion_sites$start[2]) |
         (read_end <= tTSA1_motif_insertion_sites$end[3] &
           read_end >= tTSA1_motif_insertion_sites$start[3])) %>%
  mutate(rel_cumulative_counts = NA,
         rel_counts = NA,
         read_end_gap = read_end)

tTSA1_WT_insertion_gaps <- full_length_QuantSeq_plasmids_cumulative_polyA_reads %>%
  filter(mod == "WT",
         terminator == "tTSA1",
         !URA3,
         read_end >= tTSA1_motif_insertion_sites$start[1] - 1,
         read_end <= tTSA1_motif_insertion_sites$end[3]) %>%
  mutate(rel_cumulative_counts_gap = ifelse(read_end <= tTSA1_motif_insertion_sites$end[3],
                rel_cumulative_counts[read_end == tTSA1_motif_insertion_sites$start[3] - 19],
                rel_cumulative_counts)) %>%
  mutate(rel_cumulative_counts_gap = ifelse(read_end <= tTSA1_motif_insertion_sites$start[3] - 1,
                NA,
                rel_cumulative_counts_gap)) %>%
  mutate(rel_cumulative_counts_gap = ifelse(read_end <= tTSA1_motif_insertion_sites$end[2],
                rel_cumulative_counts[ read_end == tTSA1_motif_insertion_sites$start[2] - 10],
                rel_cumulative_counts_gap)) %>%
  mutate(rel_cumulative_counts_gap = ifelse(read_end <= tTSA1_motif_insertion_sites$start[2] - 1,
                NA,
                rel_cumulative_counts_gap)) %>%
  mutate(rel_cumulative_counts_gap = ifelse(read_end <= tTSA1_motif_insertion_sites$end[1],
                rel_cumulative_counts[read_end == tTSA1_motif_insertion_sites$start[1] - 1],
                rel_cumulative_counts_gap)) 

```


# Plot cumulative poly(A)-site reads for modified terminators

```{r create-figure-QuantSeq, fig.height = 9}
RPS3_terminator_sequence = tibble(position = 1:227, 
                                  mod_NNN = str_split("ATTTAATTATTAAATACATAAATCGTCTACGAAAACTATAAGTACAAACTACGCCTTAATGCTTGAGGATTCTTCTATTCTAGTGCACTTAATTGTTGCGGTTTCTTGCATATTGCCGTGTTTAGTCAGGAGGGTGCACGTTGAAGCAGTGCGTTGAAACTAGTTGTTATTTACAGAATCACAATATTATATAATCAAAAGAAAAGTTTTGTACAACACAACTGTGA","")[[1]]) %>%
  pivot_longer(- position, names_to = "construct", values_to = "nucleotide") %>%
  mutate(colour = ifelse(construct == "WT","black",
                         ifelse(position <= tRPS3_motif_insertion_sites$end[1] &
                                position >= tRPS3_motif_insertion_sites$start[1], "white",
                         ifelse(position <= tRPS3_motif_insertion_sites$end[2] &
                                position >= tRPS3_motif_insertion_sites$start[2], "white",
                         ifelse(position <= tRPS3_motif_insertion_sites$end[3] &
                                position >= tRPS3_motif_insertion_sites$start[3], "white", "black")))))

TSA1_terminator_sequence = tibble(position = 1:246, 
                                  mod_NNN = str_split("GACGCTTGCAGAGTTGTCTAAAATCACAGAATGACTACTTATTTAAAATCAAGTTAATTCACTCTTTACGTTTATACCTATCTTATGTTGGTATCTATATATAAAACCCATAAACTTAATGCTATTCACTGACAAAACGATACACTACCGTCTAAGAAGCTATTCACACTAGCTCAGTTCTAATAACAATCATCGTCTGGCTCTAAAACGCGTAAACCATTCACTGCCACACTCACTTTAGCTCTG","")[[1]]) %>%
  pivot_longer(- position, names_to = "construct", values_to = "nucleotide") %>%
  mutate(colour = ifelse(construct == "WT","black",
                         ifelse(position <= tTSA1_motif_insertion_sites$end[1] &
                                position >= tTSA1_motif_insertion_sites$start[1], "white",
                         ifelse(position <= tTSA1_motif_insertion_sites$end[2] &
                                position >= tTSA1_motif_insertion_sites$start[2], "white",
                         ifelse(position <= tTSA1_motif_insertion_sites$end[3] &
                                position >= tTSA1_motif_insertion_sites$start[3], "white", "black")))))

pPGK1_pRPS3_cumulative_plot <- ggplot(tRPS3_WT_insertion_alpha %>%
                                        bind_rows(tRPS3_missing_values) %>%
                                        mutate(source = "QuantSeq") %>%
         filter(promoter %in% c("pRPS3", "pPGK1"), 
                mod %in% c("WT", "mod0"), 
                !URA3)) +
  geom_line(data = tRPS3_WT_insertion_gaps,
             aes(x = read_end,
                 y = rel_cumulative_counts_gap,
                 group = sample),
             colour = "red") +
  geom_vline(xintercept = c(58, 77), 
             size = 0.3) +
  cumulative_reads_plot_theme + cumulative_reads_plot_RPS3_TSA1  +
  labs(title = "tRPS3 poly(A) Sites",
       y = "Cumulative\n fraction of reads") +
  scale_x_tRPS3_mod +
  theme(plot.margin = margin(l = 20),
        axis.line.x = element_blank(),
        axis.text.x = element_blank(),
        axis.title.x = element_blank(),
        axis.ticks.x = element_blank())

pPGK1_pRPS3_construct_sequence <- ggplot() + 
  geom_blank(data = RPS3_terminator_sequence,
             aes(x = position, y = construct)) +
  geom_rect(data = tRPS3_motif_insertion_sites,
            aes(xmin = start - 0.5, xmax = end + 0.5,
                ymin = 0.75, ymax = 1.25)) + 
  geom_text(data = RPS3_terminator_sequence,
            aes(label = nucleotide, x = position, y = construct),
            colour = RPS3_terminator_sequence$colour,
            size = 2,
            family = "Mono") +
  scale_x_tRPS3_mod +
  construct_sequence_plots +
  theme(plot.margin = margin(l = 20))

pPGK1_pRPS3_cumulative_plot_with_seq <- plot_grid(pPGK1_pRPS3_cumulative_plot,
          pPGK1_pRPS3_construct_sequence,
          nrow = 2,
          align = "v",
          axis = "l",
          rel_heights = c(2,1))

pPGK1_pRPS3_cumulative_plot_all_constructs <- ggplot(tRPS3_WT_insertion_alpha %>%
                                        bind_rows(tRPS3_missing_values)  %>% 
                                        mutate(source = "QuantSeq") %>%
         filter(promoter %in% c("pRPS3", "pPGK1"), 
                mod != "WT", 
                !URA3))  +
  geom_vline(xintercept = c(58, 77), 
             size = 0.3) +
  geom_point(data = tRPS3_WT_insertion_alpha %>%
               filter(promoter %in% c("pRPS3", "pPGK1"),
                      mod != "WT",
                      !URA3,
                      read_end_gap %in% c(58, 77)),
             aes(x = read_end_gap,
                 y = rel_cumulative_counts,
                 colour = label))  +
  cumulative_reads_plot_theme + cumulative_reads_plot_RPS3_TSA1  +
  labs(title = "tRPS3 poly(A) Sites",
       x="Position relative to stop codon") +
  scale_x_tRPS3_mod +
  facet_grid(label~promoter) +
  scale_colour_manual(values = RPS3_TSA1_colour_scheme[c("WT", "mod_HTH", "mod_NTN", "mod_NAA", "mod_NNN")]) +
   theme(panel.spacing.y = unit(0.4, "cm", data = NULL))

pTSA1_cumulative_plot <- ggplot(tTSA1_WT_insertion_alpha %>%
                                        bind_rows(tTSA1_missing_values) %>% 
                                        mutate(source = "QuantSeq") %>%
         filter(promoter == "pTSA1", 
                mod %in% c("WT", "mod0"), 
                !URA3)) +
  geom_line(data = tTSA1_WT_insertion_gaps,
             aes(x = read_end,
                 y = rel_cumulative_counts_gap,
                 group = sample),
             colour = "red") +
  geom_vline(xintercept = c(127, 133), 
             size = 0.3)  +
  scale_x_tTSA1_mod +
  cumulative_reads_plot_theme + cumulative_reads_plot_RPS3_TSA1  +
  guides(colour = "none") +
  labs(y="", 
       x = "",
       title = "tTSA1 poly(A) Sites") +
  theme(plot.margin = margin(r = 20),
        axis.line.x = element_blank(),
        axis.text.x = element_blank(),
        axis.title.x = element_blank(),
        axis.ticks.x = element_blank())



pTSA1_construct_sequence <- ggplot() + 
  geom_blank(data = TSA1_terminator_sequence,
             aes(x = position, y = construct)) +
  geom_rect(data = tTSA1_motif_insertion_sites,
            aes(xmin = start - 0.5, xmax = end + 0.5,
                ymin = 0.75, ymax = 1.25)) + 
  geom_text(data = TSA1_terminator_sequence,
            aes(label = nucleotide, x = position, y = construct),
            colour = TSA1_terminator_sequence$colour,
            size = 2,
            family = "Mono") +
  scale_x_tTSA1_mod +
  construct_sequence_plots +
  theme(axis.text.y = element_blank(),
        plot.margin = margin(r = 20))

pTSA1_cumulative_plot_with_seq <- plot_grid(pTSA1_cumulative_plot,
          pTSA1_construct_sequence,
          nrow = 2,
          align = "v",
          axis = "l",
          rel_heights = c(2,1))

pTSA1_cumulative_plot_all_constructs <- ggplot(tTSA1_WT_insertion_alpha %>%
                                        bind_rows(tTSA1_missing_values) %>%
                                        mutate(source = "QuantSeq") %>%
         filter(promoter == "pTSA1",
                      mod != "WT", 
                !URA3)) +
  geom_vline(xintercept = c(127, 133), 
             size = 0.3) +
  geom_point(data = tTSA1_WT_insertion_alpha %>%
               filter(promoter == "pTSA1",
                      mod != "WT",
                      !URA3,
                      read_end_gap %in% c(127, 133)),
             aes(x = read_end_gap,
                 y = rel_cumulative_counts,
                 colour = label))  +
  scale_x_tTSA1_mod +
  cumulative_reads_plot_theme + cumulative_reads_plot_RPS3_TSA1  +
  guides(colour = "none") +
  labs(x = "Position relative to stop codon",
       title = "tTSA1 poly(A) Sites") +
  facet_grid(label~promoter) +
  theme(axis.title.y = element_blank())

quantseq_polya_site_usage_plot <- 
  plot_grid(qPCR_vs_RNASeq_plot_QuantSeq,
            plot_grid(pPGK1_pRPS3_cumulative_plot_with_seq,
                      pTSA1_cumulative_plot_with_seq),
            plot_grid(pPGK1_pRPS3_cumulative_plot_all_constructs,
                      pTSA1_cumulative_plot_all_constructs,
                      get_legend(pPGK1_pRPS3_cumulative_plot_all_constructs +
                                   guides(colour = guide_legend(ncol=1),
                                          linetype = guide_legend(ncol=1)) +
                                   theme(legend.justification = "centre",
                                         legend.direction = "vertical",
                                         legend.box = "vertical",
                                         legend.box.margin = margin(),
                                         legend.key.size = unit(0.6, 'cm'))),
                      nrow = 1,
                      rel_widths = c(2,1.3,0.6)),
            ncol = 1,
            labels = "AUTO",
            rel_heights = c(1,1.1,2.5))

quantseq_polya_site_usage_plot
```

```{r print-out-quantseq-figure}
ggsave2(here("./results_chapter/figures/polya_usage_plot_QuantSeq.png"), 
        quantseq_polya_site_usage_plot, 
        width = fig_width_2column,
        height = 210,
        units = "mm",
        dpi = fig_dpi,
        bg = "white")
```


# Plot relative counts on poly(A) sites for modified constructs

```{r relative-counts-plots, fig.height = 8}
relative_polya_counts_pRPS3_5PSeq <- ggplot(full_5PSeq_cumulative_polyA_reads %>%
         filter(!mod %in% c("genomic", "WT"),
                !URA3) %>%
  inner_join(construct_to_label_dictionary_TSA1_RPS3, by = c("mod" = "construct"))) +
  geom_step(aes(x = read_end,
                y = rel_counts,
                group = sample,
                colour = label),
            alpha = 0.7) +
  facet_wrap(~label, ncol = 3) +
  labs(title = "Relative poly(A) counts (5PSeq), pRPS3",
       x="Position relative to stop codon",
      y="Fraction of reads",
      linetype = "Promoter",
      colour = "Construct") +
  guides(linetype = "none") +
  scale_colour_manual(values = RPS3_TSA1_colour_scheme)+
  scale_x_tRPS3_mod +
  theme(legend.position = "bottom")

QuantSeq_relative_polya_counts_pPGK1_pRPS3 <- ggplot(full_QuantSeq_cumulative_polyA_reads %>%
         filter(!mod %in% c("genomic", "WT"),
                !URA3,
                terminator != "tTSA1") %>%
  inner_join(construct_to_label_dictionary_TSA1_RPS3, by = c("mod" = "construct"))) +
  geom_step(aes(x = read_end,
                y = rel_counts,
                group = sample,
                colour = label),
            alpha = 0.7) +
  facet_grid(label ~ promoter) +
  scale_x_tRPS3_mod +
  labs(x="Position relative to stop codon",
      y="Fraction of Reads",
      linetype = "Promoter",
      colour = "Construct") +
  guides(linetype = "none",
         colour = "none") +
  scale_colour_manual(values = RPS3_TSA1_colour_scheme)

QuantSeq_relative_polya_counts_pTSA1 <- ggplot(full_QuantSeq_cumulative_polyA_reads %>%
         filter(!mod %in% c("genomic", "WT"),
                !URA3,
                terminator == "tTSA1") %>%
  inner_join(construct_to_label_dictionary_TSA1_RPS3, by = c("mod" = "construct"))) +
  geom_step(aes(x = read_end,
                y = rel_counts,
                group = sample,
                colour = label),
            alpha = 0.7) +
  facet_grid(label ~ promoter) +
  labs(x="Position relative to stop codon",
      linetype = "Promoter",
      colour = "Construct") +
  guides(linetype = "none",
         colour = "none") +
  scale_colour_manual(values = RPS3_TSA1_colour_scheme) +
  scale_x_tTSA1_mod +
  theme(axis.title.y = element_blank())

relative_polya_counts_QuantSeq_title <- ggdraw() + 
  draw_label(
    "Relative poly(A) Counts (QuantSeq)",
    fontface = 'bold',
    hjust = 0.5
  )

relative_polya_counts_QuantSeq <- 
  plot_grid(relative_polya_counts_QuantSeq_title,
            plot_grid(QuantSeq_relative_polya_counts_pPGK1_pRPS3, 
                      QuantSeq_relative_polya_counts_pTSA1,
                      nrow = 1,
                      rel_widths = c(1, 0.55)),
            ncol = 1,
            rel_heights = c(0.1,1))

relative_polya_counts_QuantSeq_and_5PSeq <-
  plot_grid(relative_polya_counts_QuantSeq_title,
            plot_grid(QuantSeq_relative_polya_counts_pPGK1_pRPS3, 
                      QuantSeq_relative_polya_counts_pTSA1,
                      nrow = 1,
                      rel_widths = c(1, 0.55)),
            relative_polya_counts_pRPS3_5PSeq,
            ncol = 1,
            rel_heights = c(0.1,1,0.7))

relative_polya_counts_QuantSeq_and_5PSeq
```

```{r print-out-quantseq-5PSeq-results}

ggsave2(here("./supplementary_data_chapter/figures/relative_polya_counts_QuantSeq_and_5PSeq.png"), 
        relative_polya_counts_QuantSeq_and_5PSeq, 
        width = fig_width_2column,
        height = 210,
        units = "mm",
        dpi = fig_dpi,
        bg = "white")
```

# Compare 5PSeq reads for modified constructs

Note only tRPS3 constructs were tested with 5PSeq.

```{r create-figure-5PSeq}
tRPS3_WT_insertion_alpha_5PSeq_and_QuantSeq <- tRPS3_WT_insertion_alpha_5PSeq %>%
  bind_rows(tRPS3_missing_values_5PSeq) %>%
  mutate(source = "5PSeq") %>%
  bind_rows(tRPS3_WT_insertion_alpha %>% 
              bind_rows(tRPS3_missing_values) %>%
              filter(promoter == "pRPS3") %>%
              mutate(source = "QuantSeq"))

tRPS3_cumulative_plot_5PSeq <- ggplot(tRPS3_WT_insertion_alpha_5PSeq_and_QuantSeq %>%
         filter(promoter %in% c("pRPS3", "pPGK1"), 
                source == "5PSeq",
                mod %in% c("WT", "mod0"), 
                !URA3)) +
  geom_step(data = tRPS3_WT_insertion_gaps_5PSeq,
             aes(x = read_end,
                 y = rel_cumulative_counts_gap,
                 group = sample),
             colour = "red") +
  geom_vline(xintercept = c(58, 77), 
             size = 0.3) +
  cumulative_reads_plot_theme + cumulative_reads_plot_RPS3_TSA1  +
  labs(title = "tRPS3 poly(A) Sites",
       y = "Cumulative\n fraction of reads") +
  scale_x_tRPS3_mod +
  theme(plot.margin = margin(l = 20),
        axis.line.x = element_blank(),
        axis.text.x = element_blank(),
        axis.title.x = element_blank(),
        axis.ticks.x = element_blank())

tRPS3_cumulative_plot_with_seq_5PSeq <- plot_grid(tRPS3_cumulative_plot_5PSeq,
          pPGK1_pRPS3_construct_sequence,
          nrow = 2,
          align = "v",
          axis = "l",
          rel_heights = c(2,1))

tRPS3_cumulative_plot_all_constructs_5PSeq <- ggplot(tRPS3_WT_insertion_alpha_5PSeq_and_QuantSeq %>%
         filter(promoter %in% c("pRPS3", "pPGK1"), 
                mod != "WT", 
                source == "5PSeq",
                !URA3)) +
  geom_vline(xintercept = c(58, 77), 
             size = 0.3) +
  geom_point(data = tRPS3_WT_insertion_alpha_5PSeq %>%
               filter(promoter %in% c("pRPS3", "pPGK1"),
                      mod != "WT",
                      !URA3,
                      read_end_gap %in% c(58, 77)),
             aes(x = read_end_gap,
                 shape = promoter,
                 y = rel_cumulative_counts,
                 colour = label))  +
  cumulative_reads_plot_theme + cumulative_reads_plot_RPS3_TSA1  +
  labs(title = "tRPS3 poly(A) Sites",
       x="Position relative to stop codon",
       y = "Cumulative\n fraction of reads",
       linetype = "RNA-Seq") +
  scale_x_tRPS3_mod +
  facet_wrap(~label, nrow = 2)


polya_site_usage_plot_5PSeq <- 
  plot_grid(
    plot_grid(qPCR_vs_RNASeq_plot_5PSeq,
              tRPS3_cumulative_plot_with_seq_5PSeq, 
              nrow = 1,
              labels = c("", "B"),
              rel_widths = c(0.8,1.5)),
    plot_grid(tRPS3_cumulative_plot_all_constructs_5PSeq,
              get_legend(tRPS3_cumulative_plot_all_constructs_5PSeq +
                           guides(colour = guide_legend(nrow=2)) +
                           theme(legend.justification = "centre",
                                 legend.box.margin = margin(),
                                 legend.box = "vertical")),
              ncol = 1,
              rel_heights = c(4,1)),
    ncol = 1,
    labels = c("A", "C"),
    rel_heights = c(1,1.5))
```


# Wilcoxon test for different distributions of poly(A) sites in modified terminators and QuantSeq vs 5PSeq

```{r calc-Wilcoxon-test}
calc_Wilcoxon_test <- function(polyA_site_reads_construct, polyA_site_reads_mod_NNN){
  polyA_site_reads <- tibble(rel_counts = polyA_site_reads_construct,
                             mod_NNN = polyA_site_reads_mod_NNN)
  
  wilcox.test(polyA_site_reads %>%
                filter(!is.na(rel_counts), !is.na(mod_NNN)) %>%
                pull(rel_counts),
              polyA_site_reads %>% 
                filter(!is.na(rel_counts), !is.na(mod_NNN)) %>% 
                pull(mod_NNN),
              paired = TRUE) %>%
    tidy()
}
tRPS3_WT_polyA_site_counts <- tRPS3_WT_insertion_alpha %>%
  bind_rows(tRPS3_missing_values) %>%
  filter(promoter != "pTSA1",
         !URA3,
        read_end_gap < 170) %>% 
  replace_na(list(rel_counts = 0)) %>%
  group_by(terminator,
           promoter,
           label,
           read_end_gap) %>% 
  summarise(rel_counts = mean(rel_counts)) %>%
  group_by(terminator,
           promoter,
           read_end_gap) %>%
  mutate(mod_NNN = rel_counts[label == "mod_NNN"])

tRPS3_WT_polyA_site_counts_5PSeq <- tRPS3_WT_insertion_alpha_5PSeq %>%
  bind_rows(tRPS3_missing_values_5PSeq) %>%
  filter(promoter != "pTSA1",
         !URA3,
        read_end_gap < 170) %>% 
  replace_na(list(rel_counts = 0)) %>%
  group_by(terminator,
           promoter,
           label,
           read_end_gap) %>% 
  summarise(rel_counts_5PSeq = mean(rel_counts)) %>%
  inner_join(tRPS3_WT_polyA_site_counts) %>%
  select(-mod_NNN) %>%
  dplyr::rename(rel_counts_QuantSeq = rel_counts)

tTSA1_WT_polyA_site_counts <- tTSA1_WT_insertion_alpha %>%
  bind_rows(tTSA1_missing_values) %>%
  filter(promoter == "pTSA1",
         !URA3,
        read_end_gap < 170) %>% 
  replace_na(list(rel_counts = 0)) %>%
  group_by(terminator,
           promoter,
           label,
           read_end_gap) %>% 
  summarise(rel_counts = mean(rel_counts)) %>%
  group_by(terminator,
           promoter,
           read_end_gap) %>%
  mutate(mod_NNN = rel_counts[label == "mod_NNN"])

tTSA1_polyA_site_1_max <- tTSA1_WT_polyA_site_counts %>%
  filter(read_end_gap < 130, 
         read_end_gap >= 120) %>% 
  group_by(promoter, 
           terminator,  
           label) %>% 
  summarise(max_rel_counts = max(rel_counts, na.rm = TRUE)) %>%
  mutate(site = 1)

tTSA1_polyA_site_1 <- tTSA1_WT_polyA_site_counts %>%
  filter(read_end_gap < 130, 
         read_end_gap >= 120) %>%
  group_by(promoter,
           terminator,
           label) %>%
  summarise(calc_Wilcoxon_test(rel_counts, mod_NNN)) %>%
  mutate(site = 1)

tRPS3_polyA_site_1_max <- tRPS3_WT_polyA_site_counts %>%
  filter(read_end_gap < 65, 
         read_end_gap >= 55) %>% 
  group_by(promoter, 
           terminator,  
           label) %>% 
  summarise(max_rel_counts = max(rel_counts, na.rm = TRUE)) %>%
  mutate(site = 1)

tRPS3_polyA_site_1_max_5PSeq <- tRPS3_WT_polyA_site_counts_5PSeq %>%
  filter(read_end_gap < 65, 
         read_end_gap >= 55) %>% 
  group_by(promoter, 
           terminator,  
           label) %>% 
  summarise(max_rel_counts = max(rel_counts_5PSeq, na.rm = TRUE)) %>%
  mutate(site = 1)

tRPS3_polyA_site_1 <- tRPS3_WT_polyA_site_counts %>%
  filter(read_end_gap < 65, 
         read_end_gap >= 55) %>%
  group_by(promoter,
           terminator,
           label) %>%
  summarise(calc_Wilcoxon_test(rel_counts, mod_NNN)) %>%
  mutate(site = 1)

tRPS3_polyA_site_1_5PSeq <- tRPS3_WT_polyA_site_counts_5PSeq %>%
  filter(read_end_gap < 65, 
         read_end_gap >= 55) %>%
  group_by(promoter,
           terminator,
           label) %>%
  summarise(calc_Wilcoxon_test(rel_counts_5PSeq, rel_counts_QuantSeq)) %>%
  mutate(site = 1)

tTSA1_polyA_site_2_max <- tTSA1_WT_polyA_site_counts %>%
  filter(read_end_gap < 140, 
         read_end_gap >= 130) %>% 
  group_by(promoter, 
           terminator,  
           label) %>% 
  summarise(max_rel_counts = max(rel_counts, na.rm = TRUE)) %>%
  mutate(site = 2)

tTSA1_polyA_site_2 <- tTSA1_WT_polyA_site_counts %>%
  filter(read_end_gap < 140, 
         read_end_gap >= 130) %>%
  group_by(promoter,
           terminator,
           label) %>%
  summarise(calc_Wilcoxon_test(rel_counts, mod_NNN)) %>%
  mutate(site = 2)

tRPS3_polyA_site_2_max <- tRPS3_WT_polyA_site_counts %>%
    filter(read_end_gap < 80, 
           read_end_gap >= 70) %>%
  group_by(promoter, 
           terminator,  
           label) %>% 
  summarise(max_rel_counts = max(rel_counts, na.rm = TRUE)) %>%
  mutate(site = 2)

tRPS3_polyA_site_2_max_5PSeq <- tRPS3_WT_polyA_site_counts_5PSeq %>%
    filter(read_end_gap < 80, 
           read_end_gap >= 70) %>%
  group_by(promoter, 
           terminator,  
           label) %>% 
  summarise(max_rel_counts = max(rel_counts_5PSeq, na.rm = TRUE)) %>%
  mutate(site = 2)

tRPS3_polyA_site_2 <- tRPS3_WT_polyA_site_counts %>%
  filter(read_end_gap < 80, 
         read_end_gap >= 70) %>%
  group_by(promoter,
           terminator,
           label) %>%
  summarise(calc_Wilcoxon_test(rel_counts, mod_NNN)) %>%
  mutate(site = 2)

tRPS3_polyA_site_2_5PSeq <- tRPS3_WT_polyA_site_counts_5PSeq %>%
  filter(read_end_gap < 80, 
         read_end_gap >= 70) %>%
  group_by(promoter,
           terminator,
           label) %>%
  summarise(calc_Wilcoxon_test(rel_counts_5PSeq, rel_counts_QuantSeq)) %>%
  mutate(site = 2)

polyA_site_p.value <- tRPS3_polyA_site_1 %>%
  inner_join(tRPS3_polyA_site_1_max) %>%
  bind_rows(tTSA1_polyA_site_1 %>%
              inner_join(tTSA1_polyA_site_1_max)) %>%
  bind_rows(tRPS3_polyA_site_2 %>%
              inner_join(tRPS3_polyA_site_2_max)) %>%
  bind_rows(tTSA1_polyA_site_2 %>%
              inner_join(tTSA1_polyA_site_2_max))  %>%
  filter(label != "mod_NNN") %>%
  transmute(promoter, terminator, label, site, max_rel_counts, p.value, p.adj = p.adjust(p.value))

polyA_site_p.value_5PSeq <- tRPS3_polyA_site_1_5PSeq %>%
  inner_join(tRPS3_polyA_site_1_max_5PSeq) %>%
  bind_rows(tRPS3_polyA_site_2_5PSeq %>%
  inner_join(tRPS3_polyA_site_2_max_5PSeq)) %>%
  transmute(promoter, terminator, label, site, max_rel_counts, p.value, p.adj = p.adjust(p.value))
```

```{r print-out-5pseq-results}
write_csv(polyA_site_p.value, here("supplementary_data_chapter/data/polyA_site_p.value.csv"))

write_csv(polyA_site_p.value_5PSeq, here("supplementary_data_chapter/data/polyA_site_p.value_5PSeq.csv"))

ggsave2(here("supplementary_data_chapter/figures/polya_usage_plot_5PSeq.png"), 
        polya_site_usage_plot_5PSeq, 
        width = fig_width_2column,
        height = 150,
        units = "mm",
        dpi = fig_dpi,
        bg = "white")
```
