---
output:
  pdf_document: default
  html_document: default
---

# Discussion

Our measurements of protein output from promoter-terminator combinations highlight the limitations of assuming that CREs are quantitatively composable.
Unsurprisingly, fluorescent protein expression levels are dominated by promoter choice, up to 100 fold, and then terminator choice, up to 5 fold. 
Our observations of 1.5 fold change in the relative effect of terminators depending on coding sequence and promoter choice highlight the quantitative limitations of the assumption of composability.
Our results echo previous findings of idiosyncratic and context-dependent interactions between CREs @Dhillon2020.
Previous work by Kosuri et al. @Kosuri2013 suggests that screening a larger library of CREs would reveal a subset with stronger quantitative interactions.
These limitations in composability mean that combinations of CREs may work unpredictably in synthetic gene constructs, and so that genetic engineers should routinely consider testing multiple combinations of CREs.

We focused on quantifying the context dependent behaviour of short cis-regulatory sequence motifs within the 3â€™UTR, in multiple promoter and terminator contexts.
The contribution of our selected  motifs to mRNA stability, as inferred from transcriptome-wide analysis, correlates with their effect on the abundance of reporter constructs.
However, their contributions to mRNA abundance were inconsistent and dependent on context: TGTAHMNTA has no measurable effect when inserted into tRPS3 but the expected effect in tTSA1, ATATTC has the expected effect in tRPS3 but little effect in tTSA1, and HWNCATTWY can either decrease or increase mRNA levels when removed from tPIR1, depending on the promoter.
Also, the two tPIR1 constructs with different mutated HWNCATTWY motifs had different expression levels suggesting HWNCATTWY has a position-dependent effect. 
The exact sequences of the two HWNCATTWY motif instances did also differ by 4 nucleotides.
These disparities in predicted and experimentally verified regulatory behaviour raises questions about current methods for discovering novel cis-regulatory elements and, more generally, for understanding and manipulating regulatory pathways.

The effect of CREs can also depend on environmental context.
Our motif search predicted a TGTAAATA motif to be stabilising in the Sun et al. @Sun2013 dataset but destabilising in the Chan et al. @Chan2018 dataset.
TGTAAATA is a known binding motif for Puf3, and Puf3 activity in promoting mRNA degradation or translation depends on growth conditions including carbon source [@Miller2014; @Lee2015puf3].
Growth conditions differed between the assays, with rich YPD media used by @Sun2013 and bespoke CSM-lowURA media used by @Chan2018, more similar to the SC-URA used in out experiments.
Although we are not aware of direct experimental comparisons of Puf3 effects beyond specifically manipulating the carbon source, nutrient-dependent regulation by Puf3 might explain the different behaviour of TGTAAATA between datasets.

Our poly(A) mapping results show that poly(A) site usage is context independent, as long as sequence elements determining termination are unaltered, in agreement with Lui et al. [@lui2022].
Poly(A) site usage is remarkably consistent across terminators that are placed on plasmids in different promoter and coding sequence contexts that are different from the native genomic context (Figure \@ref(fig:quantseq-polyA-site-usage)).
The sequences and relative usages of the poly(A) sites are not affected by inserting motifs into tTSA1 constructs.
For tRPS3 constructs, poly(A) sites are again appropriately shifted downstream by the length of the insertions.
However, a novel poly(A) site is introduced between the second and third inserted motifs.
A possible explanation for the novel poly(A) site is that motifs 2 and 3 were inserted into a evolutionarily conserved element of tRPS3, as detected by pHastCons [@Siepel2005].
Insertion might have disrupted an RBP target site or secondary structure in this region, leading to changes in polyadenylation site choice.

Our work quantifying the context-dependent contributions of individual short sequence motifs challenges working models that assume that motifs have additive independent effects.
Other lines of evidence challenging assumptions of composability include: regulatory protein binding to motifs in mRNA depends on the sequence context [@Taliaferro2016]; some regulatory RNA-binding proteins bind with gapped multi-partite motifs [@Bayne2021]; and some motifs must be repeated in the same transcript to be detectably functional [@Gu2004;@Jackson2004].
A combination of experimental and computational techniques, as used here, enables the quantification of more complex context-dependent regulatory effects. 

Conceptual models of genes are built in to the tools that biologists use, and can limit their ability to predict and manipulate gene expression [@Szymanski2019].
Working models of CREs as independently acting composable parts exemplify these limits.
The characterisation of an element's function in a single context may not lead to generalisable conclusions.
To build a comprehensive understanding of transcriptional and post-transcriptional regulation, studies of CREs should ideally be carried out with multiple combinations of other regulatory elements.


# Funding

This work was supported by the Wellcome Trust [grant number 208779/Z/17/Z to E.W.J.W.], the UKRI Biotechnology and Biological Sciences Research Council [grant number BB/M010996/1], and the Medical Research Council [grant number MR/N013166/1].

# Data Availability

Plate reader data, RT-qPCR data, and complete analysis code is available at https://github.com/DimmestP/chimera_project_manuscript.
RNA-seq data is available on Gene Expression Omnibus [GSE200211](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE200211).
Complete code for RNA-seq data analysis is at https://github.com/DimmestP/nextflow_paired_reads_pipeline.

# Acknowledgments

We thank Naomi Nakayama, Erika Szymanski, Filippo Menolascina, and members of the Wallace lab for discussions on the manuscript.
We thank Ivan Clark, Iseabail Farquhar, Luis Montano and Peter Swain for help with the plate reader experiments and analysis.
We thank Vicent Pelechano and Yujie Zhang for their generous help teaching us the 5PSeq protocol, obtaining 5PSeq data, and discussing the results.
We thank the Genetics Core, Edinburgh Clinical Research Facility, University of Edinburgh for their expertise and assistance obtaining RNA-seq data.

# References

::: {#refs}
:::

\newpage
